### PR TITLE
AdmStorm 2.0: @admstorm-marker

### PR DESCRIPTION
**Pull request by Danil42Russia**

### CODE REVIEW

File: src/main/kotlin/com/vk/admstorm/tags/phptags/psi/AdmStormMarkerItemPsiImpl.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+class AdmStormMarkerItemPsiImpl(node: ASTNode) : PhpDocPsiElementImpl(node) {
+    companion object {
+        val elementType = PhpDocElementType("admStormMarkerItemPsiImpl")
+    }
+
+    private val value = node.text.split(" ", limit = 2)
+
+    fun kindValue(): String {
+        return value[0].removePrefix("kind=")
```
Comment (i582) [inline src/main/kotlin/com/vk/admstorm/tags/phptags/psi/AdmStormMarkerItemPsiImpl.kt:15]: maybe `...(): String = value[0].removePrefix("kind=")`

File: src/main/kotlin/com/vk/admstorm/utils/DslBuilderUtils.kt [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package com.vk.admstorm.utils
+
+import com.intellij.ui.dsl.builder.Cell
+import com.intellij.util.ui.JBUI
+import java.awt.Font
+import javax.swing.JComponent
+
+object DslBuilderUtils {
```
Comment (i582) [inline src/main/kotlin/com/vk/admstorm/utils/DslBuilderUtils.kt:8]: `UiDslBuilderUtils`

**Review by i582 — APPROVED** [review]

File: src/main/kotlin/com/vk/admstorm/tags/phptags/psi/AdmStormMarkerItemPsiImpl.kt (hunk 1 above)
Comment (Danil42Russia) [reply src/main/kotlin/com/vk/admstorm/tags/phptags/psi/AdmStormMarkerItemPsiImpl.kt]: done

File: src/main/kotlin/com/vk/admstorm/utils/DslBuilderUtils.kt (hunk 2 above)
Comment (Danil42Russia) [reply src/main/kotlin/com/vk/admstorm/utils/DslBuilderUtils.kt]: done

File: src/main/kotlin/com/vk/admstorm/highlight/markers/admmarker/LoggerMarker.kt [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+import com.vk.admstorm.env.Env
+import com.vk.admstorm.ui.AdmIcons
+
+class LoggerMarker : MarkerBrowser() {
+    override fun getIcon() = AdmIcons.General.ExternalLinkArrow
+
+    override fun getTooltip() = "Go to show log"
+
+    override fun generateUrl(uri: String): String {
+        val url = Env.data.services.getByKey("ch_logger") ?: return ""
```
Comment (i582) [inline src/main/kotlin/com/vk/admstorm/highlight/markers/admmarker/LoggerMarker.kt:12]: maybe `service`?

File: src/main/kotlin/com/vk/admstorm/highlight/markers/admmarker/MarkerBrowser.kt [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package com.vk.admstorm.highlight.markers.admmarker
+
+import com.vk.admstorm.env.Service
+
+abstract class MarkerBrowser : IMarkerBrowser {
+    fun List<Service>.getByKey(key: String): Service? {
+        return this.firstOrNull { it.key == key }
```
Comment (i582) [inline src/main/kotlin/com/vk/admstorm/highlight/markers/admmarker/MarkerBrowser.kt:7]: maybe `= firstOrNull { it.key == key }`?

File: src/main/kotlin/com/vk/admstorm/highlight/markers/admmarker/StatshouseMarker.kt [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    override fun getIcon() = AdmIcons.General.ExternalLinkArrow
+
+    override fun getTooltip() = "Go to statshouse"
+
+    enum class Mode(val modeName: String) {
+        VIEW("view"),
+    }
+
+    override fun generateUrl(uri: String): String {
+        val url = Env.data.services.getByKey("statshouse") ?: return ""
```
Comment (i582) [inline src/main/kotlin/com/vk/admstorm/highlight/markers/admmarker/StatshouseMarker.kt:16]: maybe `service`

**Review by i582 — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/vk/admstorm/highlight/markers/admmarker/LoggerMarker.kt (hunk 3 above)
Comment (Danil42Russia) [reply src/main/kotlin/com/vk/admstorm/highlight/markers/admmarker/LoggerMarker.kt]: done

File: src/main/kotlin/com/vk/admstorm/highlight/markers/admmarker/MarkerBrowser.kt (hunk 4 above)
Comment (Danil42Russia) [reply src/main/kotlin/com/vk/admstorm/highlight/markers/admmarker/MarkerBrowser.kt]: done

File: gradle.properties [hunk 6]
```diff
@@ -5,7 +5,7 @@ pluginName=admstorm
 pluginRepositoryUrl = https://github.com/VKCOM/admstorm
 
 # SemVer format -> https://semver.org
-pluginVersion=1.3.5
+pluginVersion=2.0.0-beta.2
```
Comment (unserialize) [inline gradle.properties:8]: 2.0.0

Comment (Danil42Russia) [reply gradle.properties]: yep, forgot to push the commit

**Review by unserialize — APPROVED** [review]

**Pull request merged**